### PR TITLE
fix embed page by making it's dependencies very odd

### PR DESCRIPTION
**Pull request by jamesgorrie**
## What does this change?

Currently we have a few different ways to of instantiating our player.
This seems to have broken a few things over the multiple iterations of things.

Omniture tracking on our embeds is one of those things - which we report to ABC.

This fix is ugly - the first implementation if the parent of that. 

I will be using this as an instigator to fix it properly, but ABC have come back to use saying, "Do we, or don't we include these" - which is why the hack.
## What is the value of this and can you measure success?

We start tracking starts off platform again.
## Does this affect other platforms - Amp, Apps, etc?

No
## Screenshots

:rage4: <= me
## Request for comment

@akash1810 @mkopka @sndrs

### CODE REVIEW

File: applications/app/views/videoEmbed.scala.html [hunk 1]
```diff
@@ -69,7 +69,7 @@
                 )
             ),
             config: @Html(templates.js.javaScriptConfig(model.SimpleContentPage(video)).body),
-            adBlockers: {}
+            adBlockers: { onDetect: [] }
```
Comment (jamesgorrie) [inline applications/app/views/videoEmbed.scala.html]: This was added as we call `.push` on it elsewhere.

File: applications/app/views/videoEmbed.scala.html [hunk 2]
```diff
@@ -97,11 +97,14 @@
...
-            window.curl.paths['bootstraps/video-embed'] = '@Static("javascripts/bootstraps/video-embed.js")';
-            window.curl.paths['ophan/embed'] = '@{Configuration.javascript.config("ophanEmbedJsUrl")}';
-
+            @if(Configuration.assets.useHashedBundles) {
+                window.curl.paths['bootstraps/video-embed'] = '@Static("javascripts/bootstraps/video-embed.js")';
+                window.curl.paths['ophan/embed'] = '@{Configuration.javascript.config("ophanEmbedJsUrl")}';
+            } else {
+                window.curl.paths['bootstraps/video-embed'] = '@{Configuration.assets.path}javascripts/bootstraps/video-embed.js';
+                window.curl.paths['ophan/embed'] = '@{Configuration.javascript.config("ophanEmbedJsUrl")}';
+            }
```
Comment (jamesgorrie) [inline applications/app/views/videoEmbed.scala.html]: ^ This is replicating what we do in `curlConfig.scala.js`

File: common/app/templates/inlineJS/blocking/curlConfig.scala.js [hunk 3]
```diff
@@ -65,7 +65,8 @@ window.curlConfig = {
             'bootstraps/enhanced/media/video-player':   'bootstraps/enhanced/media/video-player-dev.js',
             videojs:                                    'components/video.js/video.js',
             'videojs-contrib-ads':                      'components/videojs-contrib-ads/videojs.ads.js',
-            videojsembed:                               'components/videojs-embed/videojs.embed.js',
+            videojsembed:                               'bootstraps/enhanced/media/videojsembed-dev.js',
+            videojsembeddev:                            'components/videojs-embed/videojs.embed.js',
```
Comment (jamesgorrie) [inline common/app/templates/inlineJS/blocking/curlConfig.scala.js]: Ah, self referential. Better that creating a whole `video-embed-dev` [like we did for video-player](https://github.com/guardian/frontend/blob/master/static/src/javascripts/bootstraps/enhanced/media/video-player-dev.js).

Comment (sndrs) [inline common/app/templates/inlineJS/blocking/curlConfig.scala.js]: is this only used in dev? seems very wrong to have a bootstrap file in the source that's only used in dev?

Comment (jamesgorrie) [inline common/app/templates/inlineJS/blocking/curlConfig.scala.js]: Yip, but that's the way we've chosen to do it.
As mentioned this is a hackfix to get our tracking back.

**Comment by akash1810** [comment]
😿 but 👍

**Comment by jamesgorrie** [comment]
Indeed - watch this space.

**Comment by mkopka** [comment]
Unfortunately this hasn't brought the data back on-line James

On 29 June 2016 at 15:02, James Gorrie <notifications@github.com> wrote:

> Indeed - watch this space.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/guardian/frontend/pull/13433#issuecomment-229364990>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGP_2Hq4Emh-HfHrj7a3hH6g_dqjl8MFks5qQnrbgaJpZM4JBFLX>
> .
>



-- 

Thanks - Matt


Kings Place, 90 York Way,

London N1 9GU
theguardian.com

-- 

------------------------------
This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396

**Pull request merged**